### ISSUE TITLE
LaTeX support 

### ISSUE THREAD

**Issue by kgv** [lampsitter/egui_commonmark#25]
Is LaTeX support planned?

**Comment by lampsitter** [comment]
LaTeX in general i probably too much for this crate. With that said, a subset for math would definitely be nice to have.

**Comment by zeozeozeo** [comment]
seems like pulldown_cmark has merged the support for math expressions: https://github.com/pulldown-cmark/pulldown-cmark/pull/734
we could then use [MathJax](https://docs.rs/mathjax/latest/mathjax/) to render it to an SVG and draw it

**Comment by lampsitter** [comment]
It would be really nice. It's just incredibly painful that node or headless chrome must be involved. I would have preferred to avoid that.

**Comment by zeozeozeo** [comment]
> It would be really nice. It's just incredibly painful that node or headless chrome must be involved. I would have preferred to avoid that.

True, I guess we could embed a JavaScript engine like QuickJS or Duktape and make MathJax run in it? :P
I haven't found any pure Rust TeX -> SVG renderer, aside from [Tectonic](https://github.com/tectonic-typesetting/tectonic) which seems to only render to HTML

**Comment by lampsitter** [comment]
Yeah. Your first suggestion is the most sane option. Maybe this gives me an excuse to add support for [mermaid](http://mermaid.js.org/intro/) too :thinking: 

There is definitely a lot to consider. For instance the headless chrome feature in that crate fetches mathjax from CDN and as I would expect most people to use that backend it would mean that it would require internet access to work. At least that would have to be documented.

**Comment by zeozeozeo** [comment]
or we can just leave it to the user and make then a closure that will render LaTeX into a texture

**Comment by lampsitter** [comment]
I don't think moving the problem to the user is the solution :laughing:

**Comment by lampsitter** [comment]
Looking further, I guess it's fine with the node backend. It would just require a fair bit of build instructions.

**Comment by zeozeozeo** [comment]
> I don't think moving the problem to the user is the solution :laughing:

I think that's the best option, since there's no easy way of rendering LaTeX in Rust. The implementation would also be different when compiling for web, since there you can directly interface with the browsers' JS engine

**Comment by lampsitter** [comment]
> I think that's the best option, since there's no easy way of rendering LaTeX in Rust. The implementation would also be different when compiling for web, since there you can directly interface with the browsers' JS engine

True. I did not think about for web.
